packages/flutter_tools/lib/src/build_system/targets/assets.dart

```diff
@@ -57,6 +94,128 @@ Future<Depfile> copyAssets(Environment environment, Directory outputDirectory) a
   return Depfile(inputs, outputs);
 }
 
+/// Returns a map of [FontSubsetData] keyed by relative path.
+Future<Map<String, FontSubsetData>> _getIconData(Environment environment, DevFSStringContent fontManifest) async {
```

Comment:
Can we collect the various private methods here into a single class? Once this is done, it would be better if injected values like the filesystem, artifacts, or logger came from the class instead. For example, see https://github.com/flutter/flutter/pull/48444/files